src/extractor/guidance/intersection_handler.cpp

```diff
@@ -60,7 +59,14 @@ TurnType::Enum IntersectionHandler::findBasicTurnType(const EdgeID via_edge,
     if (!on_ramp && onto_ramp)
         return TurnType::OnRamp;
 
-    if (in_data.name_id == out_data.name_id && in_data.name_id != EMPTY_NAMEID)
+    const auto same_name =
+        !util::guidance::requiresNameAnnounced(name_table.GetNameForID(in_data.name_id),
+                                               name_table.GetRefForID(in_data.name_id),
+                                               name_table.GetNameForID(out_data.name_id),
+                                               name_table.GetRefForID(out_data.name_id),
+                                               street_name_suffix_table);
+
+    if (in_data.name_id != EMPTY_NAMEID && same_name)
```

Comment:
the above check (by transitivity) checked for `in_data.name_id == out_data.name_id && in_data.name_id != EMPTY_NAME_ID && out_data.name_id != EMPTY_NAME_ID`. The final check of `out_data.name_id != EMPTY_NAME_ID` is missing now.